java/connectors/semantickernel-connectors-memory-sqlite/src/main/java/com/microsoft/semantickernel/connectors/memory/sqlite/SqliteMemoryStore.java

```diff
@@ -165,17 +164,48 @@ public Mono<Void> removeBatchAsync(
     }
 
     @Override
-    public Mono<Collection<Tuple2<MemoryRecord, Number>>> getNearestMatchesAsync(
+    public Mono<Collection<Tuple2<MemoryRecord, Float>>> getNearestMatchesAsync(
             @Nonnull String collectionName,
             @Nonnull Embedding embedding,
             int limit,
             double minRelevanceScore,
             boolean withEmbeddings) {
-        return null;
+
+        Mono<List<Database.DatabaseEntry>> entries =
+                this.dbConnector.readAllAsync(this.dbConnection, collectionName);
+
+        return entries.flatMap(
+                databaseEntries -> {
+                    EmbeddingVector embeddingVector = embedding.getVector();
+                    List<Tuple2<MemoryRecord, Float>> results = new ArrayList<>();
+                    for (Database.DatabaseEntry entry : databaseEntries) {
+                        if (entry.getEmbedding() == null || entry.getEmbedding().isEmpty()) {
+                            continue;
+                        }
+                        try {
+                            Embedding recordEmbedding =
+                                    new ObjectMapper()
+                                            .readValue(entry.getEmbedding(), Embedding.class);
+                            float relevanceScore = embeddingVector.cosineSimilarity(recordEmbedding.getVector());
+                            if (relevanceScore >= minRelevanceScore) {
+                                MemoryRecord record =
+                                        MemoryRecord.fromJsonMetadata(
+                                                entry.getMetadata(),
+                                                recordEmbedding,
+                                                entry.getKey(),
+                                                entry.getTimestamp());
+                                results.add(Tuples.of(record, relevanceScore));
+                            }
+                        } catch (JsonProcessingException e) {
+                            throw new RuntimeException(e);
```

Comment:
Is there a more specific exception for this?

Reply:
changed to throw SQLConnectorException with text along the lines of "cannot deserialize database entry".